client/src/featureform/resources.py

```diff
@@ -1403,13 +1408,35 @@ class SparkCredentials:
     def __init__(self,
                  master: str,
                  deploy_mode: str,
+                 python_version: str,
                 ):
 
         if deploy_mode != "cluster" and deploy_mode != "client":
             raise Exception(f"Spark does not support '{deploy_mode}' deploy mode. It only supports 'cluster' and 'client'.")
 
+        python_version = self._verify_python_version(python_version)
```

Comment:
Isn't this only necessary for client mode? Couldn't we get this info from the spark cluster? Could we make this default to the user's current version?

Reply:
yes, only needed it for client mode. We will need access to spark cluster in order to determine the type. The user's current version can be different from the cluster version.

Reply:
I know, it just seems a bit weird to have to add the python version of spark as a parameter. I was saying that we could default to the client's current version, if they don't specify

Is there no way to deduce the python version from Spark itself on the backend

Reply:
So we can make it _required_ if the run it in client mode. Even then if they don't specify it, we can make pick their python version on the client.